utils/build/docker/runner.Dockerfile

```diff
@@ -0,0 +1,31 @@
+FROM python:3.9
+
+
+RUN mkdir /app
+WORKDIR /app
+
+COPY requirements.txt .
+COPY build.sh .
+COPY utils/build/build.sh utils/build/build.sh
+RUN mkdir -p /app/utils/build/docker && ./build.sh -i runner
+
+# tests
+COPY tests /app/tests
+COPY parametric /app/parametric
+
+# basically everything except utils/build
+COPY utils/_context /app/utils/_context
+COPY utils/assets /app/utils/assets
+COPY utils/grpc /app/utils/grpc
+COPY utils/proxy /app/utils/proxy
+COPY utils/interfaces /app/utils/interfaces
+COPY utils/scripts /app/utils/scripts
+COPY utils/test_the_test /app/utils/test_the_test
+COPY utils/*.py /app/utils/
+
+# toplevel things
+COPY conftest.py /app/
+COPY pyproject.toml /app/
+COPY run.sh /app/
```

Comment:
This section attempts to minimize dependencies when building the image. In this case `utils/build` is entirely irrelevant but it can't be added to `.dockerignore` since the latter has its own docker things.

A bit of refactoring with dedicated folders might make that easier.

Reply:
I would say you want to have all of this as volume, to avoid rebuilding the runner each time you modify a test, WDYT ?

Reply:
Nothing about this prevents anyone to `-v $PWD:/app`, the bind mount will mask the image directory should one want to use docker for development.

This makes the image entirely self contained, which is what we want e.g in CI for caching, reproducibility, and consistency, and generally the whole purpose of a docker image.

Pushed to an extreme, with the new runner handling all the container spawning it makes it very possible to run the system tests suite without ever checking out the system tests repository.